cardinal/ecs/component.go

```diff
@@ -12,6 +13,10 @@ import (
 	"pkg.world.dev/world-engine/cardinal/ecs/filter"
 )
 
+type IGettableRawJsonFromEntityId interface {
+	GetRawJson(w *World, id entity.ID) (json.RawMessage, error)
+}
```

Comment:
Include this:

```
var _ IGettableRawJsonFromEntityId = &ComponentType[int]{}
```

To ensure at compile time that the ComponentType implements that interface.

Reply:
Will do. Question: I don't completely the risk here if I don't do this? because it did pass the tests without this.

Reply:
the risk from not doing it isn't as large in this case - however its a nice visual indicator as to what type implements the interface.